src/main/java/seedu/address/model/delivery/Id.java

```diff
@@ -0,0 +1,34 @@
+package seedu.address.model.delivery;
+
+public class Id {
+
+    public static final String MESSAGE_CONSTRAINTS = "Id should be integer";
+    private static int count = 0;
+
+    public String value;
+
+    public Id() {
+        this.value = String.valueOf(Id.count);
+        Id.count += 1;
+    }
+
+    public Id(String id) {
+        this.value = id;
+        int intId = Integer.parseInt(id);
+        count = Math.max(intId, count);
+    }
```

Comment:
This could cause multiple deliveries to have the same ID. Perhaps this constructor could be removed if it is not really essential?